app/assets/javascripts/components/campaign/campaign.jsx

```diff
@@ -0,0 +1,52 @@
+// /* eslint no-undef: 2 */
+import { Route, Switch } from 'react-router-dom';
+import { withRouter } from 'react-router';
+import React from 'react';
+import createReactClass from 'create-react-class';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import CampaignNavbar from '../common/campaign_navbar.jsx';
+import { getCampaign } from '../../actions/campaign1_actions';
+import CampaignAlerts from '../alerts/campaign_alerts.jsx';
+import CampaignOresPlot from './campaign_ores_plot.jsx';
+
+export const Campaign = createReactClass({
+  displayName: 'Campaign',
+
+  propTypes: {
+    campaign: PropTypes.object.isRequired,
+    match: PropTypes.object,
+  },
+
+  componentDidMount() {
+    const campaignSlug = this.props.match.params.slug;
+    return this.props.getCampaign(campaignSlug);
+  },
+
+  render() {
+    if (!this.props.campaign.uploads_in_use_count) {
+      return <div />;
+    }
+    return (
+      <div>
+        <CampaignNavbar
+          campaign={this.props.campaign}
+        />
+        <Switch>
+          <Route exact path="/campaigns/:campaign_slug/ores_plot" render={() => <CampaignOresPlot campaign={this.props.campaign} />} />
+          <Route exact path="/campaigns/:campaign_slug/alerts" render={() => <CampaignAlerts campaign_slug={this.props.campaign.slug} />} />
```

Comment:
The error in the alert tab is from not receiving the `match` prop as expected. You can fix that by using `component=` instead of a render function here:
`<Route exact path="/campaigns/:campaign_slug/alerts" component={CampaignAlerts} />`

Reply:
I'd rather suggest using the `render` as doing it in component way gives returns different props, 
she can follow same method/ procedure used in `course.jsx` that is either defined some variable which contains required props and pass them

Reply:
I'm fine with either way. The component way has the main advantage that it retains the interface for `CampaignAlerts` without requiring any changes, but the render way is more explicit and perhaps easier to understand in terms of data flow.